tests/adapters/test_adapter.py

```diff
@@ -62,15 +110,33 @@ def test_not_implemented_methods(adapter, method, args):
                           ))
 def test_adapter_values(value, kwargs, result):
     a = FakeAdapter()
-    assert a.values(value, **kwargs) == result
+    with pytest.warns(FutureWarning):
+        assert a.values(value, **kwargs) == result
+
+
+def test_read_binary_values():
+    a = ProtocolAdapter([(None, "abcdefgh")])
+    assert list(a.read_binary_values()) == pytest.approx([1.6777999e+22, 4.371022e+24])
```

Comment:
The what now? This would profit from more sensible input values, I think, so one can understand how those floats are supposed to be generated? Or compare the binary values directly? Or what are you trying to do here?

Reply:
I just try to verify, that some communication turns correctly to values. I changed it to the examples from numpy.